modules/dnn/perf/perf_convolution1d.cpp

```diff
@@ -0,0 +1,166 @@
+        const Conv1DParam_t& p = testConvolution1DConfigs[(int)v];
+
+        *os << "GFLOPS=" << cv::format("%.3f", p.declared_flops * 1e-9)
+            << ", K=[" << p.kernel << "]"
+            << ", IN={" << p.shapeIn.dims[0] << ", " << p.shapeIn.dims[1] << ", " << p.shapeIn.dims[2] << "}"
+            << ", OCN=" << p.outCN;
+        if (p.groups > 1)
+            *os << ", G=" << p.groups;
+        if (p.stride != 1)
+            *os << ", S=" << p.stride;
+        if (p.dilation != 1)
+            *os << ", D="  << p.dilation;
+        if (p.pad[0] != 0 && p.pad[1] != 0 )
+            *os << ", P=(" << p.pad[0] << ", " << p.pad[1] << ")";
+        if (!((std::string)p.padMode).empty())
+            *os << ", PM=" << ((std::string)p.padMode);
+        if (p.hasBias)
+            *os << ", BIAS";
+    }
+
+
+    typedef tuple<Conv1DParamID, tuple<Backend, Target> > Conv1DTestParam_t;
+    typedef TestBaseWithParam<Conv1DTestParam_t> Conv1D;
+
+    PERF_TEST_P_(Conv1D, conv1d)
+{
+    int test_id = (int)get<0>(GetParam());
+    ASSERT_GE(test_id, 0); ASSERT_LT(test_id, Conv1DParamID::CONV_LAST);
+    const Conv1DParam_t& params = testConvolution1DConfigs[test_id];
+    double declared_flops = params.declared_flops;
+
+    DictValue kernel   = DictValue::arrayInt(&params.kernel, 1);
+    DictValue stride   = DictValue::arrayInt(&params.stride, 1);
+    DictValue pad      = DictValue::arrayInt(&params.pad[0], 2);
+    DictValue dilation = DictValue::arrayInt(&params.dilation, 1);
+
+    MatShape inputShape = MatShape(params.shapeIn.dims, params.shapeIn.dims + 3);
+    int outChannels = params.outCN;
+    int groups = params.groups;
+    std::string padMode(params.padMode);
+
+    bool hasBias = params.hasBias;
+    Backend backendId = get<0>(get<1>(GetParam()));
+    Target targetId = get<1>(get<1>(GetParam()));
+
+    if (targetId != DNN_TARGET_CPU)
+    throw SkipTestException("Only CPU is supported");
+
+    int inChannels = inputShape[1];
+
+    int sz[] = {outChannels, inChannels / groups, params.kernel};
+    Mat weights(3, &sz[0], CV_32F);
+    randu(weights, -1.0f, 1.0f);
+
+    LayerParams lp;
+    lp.set("kernel_size", kernel);
+    lp.set("pad", pad);
+    if (!padMode.empty())
+    lp.set("pad_mode", padMode);
+
+    lp.set("stride", stride);
+    lp.set("dilation", dilation);
+    lp.set("num_output", outChannels);
+    lp.set("group", groups);
+    lp.set("bias_term", hasBias);
+    lp.set("is_1d", true);
+    lp.type = "Convolution";
+    lp.name = "testLayer";
+    lp.blobs.push_back(weights);
+
+    if (hasBias)
+{
+    Mat bias(1, outChannels, CV_32F);
+    randu(bias, -1.0f, 1.0f);
+    lp.blobs.push_back(bias);
+}
+int inpSz[] = {1, inChannels, inputShape[2]};
+Mat input(3, &inpSz[0], CV_32F);
+randu(input, -1.0f, 1.0f);
+
+Net net;
+net.addLayerToPrev(lp.name, lp.type, lp);
+
+net.setInput(input);
+net.setPreferableBackend(backendId);
+net.setPreferableTarget(targetId);
+
+Mat output = net.forward();
```

Comment:
Please add some check for output result. At least output shape.

Reply:
Actually, this is a warmup iteration, I don't think it is necessary to check the correctness of output shapes inside performance tests (it is already covered in other tests)